// prehtml_dqc/index.js
var fs = require('fs');


var file_prepros = process.argv[2];
var option = process.argv[3];

function html(fileName){
	var file_tmp = []
	
	fs.readFile(fileName+'.plang','utf8',function(error,data){

		var lineas = data.split('\n');
		
		lineas.forEach(function(line){
			var tag = processLine(line);
			var tag_pre = prepros(tag);
			file_tmp.push(tag_pre);
		})

		fs.writeFile(fileName+'.html',file_tmp.join('\n'));

	});	
}


function prepros(tag){

	var attributes = [''];

	if(tag.id != ''){
		attributes.push('id="'+tag.id+'"')
	}

	if(tag.className != ''){
		attributes.push('class="'+tag.className+'"')
	}

	return '<'+tag.tag+attributes.join(' ')+'>'+tag.content+'</'+tag.tag+'>';
}

function processLine(line){
	var base_tag = line.split(' ');
	
	var result = {
		tag: '',
		content: base_tag[1],
		id: '',
		className: ''
	}

	if(base_tag[0].indexOf('#') > -1){
		var tag_tmp = base_tag[0].split('#');
		result.tag = tag_tmp[0];
		result.id = tag_tmp[1];
	}else if(base_tag[0].indexOf('.') > -1){
		var tag_tmp = base_tag[0].split('.');
		result.tag = tag_tmp[0];
		result.className = tag_tmp[1];
	}else {
		result.tag = base_tag[0];
	}

	return result;
}


if( option == '-w'){
	fs.watchFile(file_prepros+'.plang',function(){
		html(file_prepros);
	})
}else{
	html(file_prepros);
}
exports.prepos = html
